src/neuroglancer/ui/layer_side_panel_state.ts

```diff
@@ -70,7 +70,10 @@ export class UserLayerSidePanelState extends RefCounted {
     for (const id of this.tabs) {
       const {hidden} = this.layer.tabs.options.get(id)!;
       if (hidden) {
-        this.registerDisposer(hidden.changed.add(() => this.panels.updateTabs()));
+        this.registerDisposer(hidden.changed.add(() => {
+          this.panels.updateTabs();
+          this.tabsChanged.dispatch();
```

Comment:
I had to add tabsChanged.dispatch because otherwise the tab label was not appearing on refresh if the graph tab was alone in its own panel.

I also noticed an issue where turning off the graph source would cause the graph tab to stay selected even though the tab content and tab label went away.

I tried to fix it by adding this code in TabView.makeTabs but then I got the unwanted behavior of the graph tab being deselected on reload since the tab isn't immediately available. Seems like that could be a tricky fix without adding complexity.
```ts
// swap to the first tab if the selected tab becomes hidden
    const tabIds = this.tabs.value.map(x => x.id);
    const selectedTabId = this.selectedTab.value;
    if (selectedTabId && tabIds.length && !tabIds.includes(selectedTabId)) {
      this.selectedTab.value = tabIds[0];
    }
```

Reply:
another issue I noticed in regards to disabling the graph source is that the tool keybind stays active and the multicut tool is assuming there is a graph source. I can handle that in the tool, is that what you would suggest? Or something similar to the DependentView, a DependentTool.

Reply:
I think for consistency with how reloading is handled, it makes sense that disabling the graph source should not deselect the graph tab if it is already selected.

I would say that if a hidden tab is selected, its tab label should also be shown (to avoid confusion).

Similarly any key bindings for tools that require a graph should also stay, but the tool itself should not work.  If the multicut tool is actually activated, rather than just bound to a key, if the graph source is disabled then I would say the tool should become deactivated (but still remain as a key binding).  I think it would be reasonable to just handle this behavior in the tool itself for now, though if you think it would be cleaner to factor out the logic somehow that would also be fine.

Reply:
To be clear, if a "hidden" tab is *not* selected then its label should be hidden.  But if a "hidden" tab is selected, then its label should be shown.  The content hiding doesn't necessarily need to be implemented by `tab_view` itself since the SegmentationGraphSourceTab can already handle that with its DependentViewWidget.

Reply:
Does this seem right? https://github.com/google/neuroglancer/pull/395/commits/6248302d75d8d590997a2476fe77b39b021150ee
I did have to listen to the selectedTab event to hide the tab if they select off of a hidden tab. I could also add a check to only redo the tabs if the previous value was a hidden tab.

Reply:
Thanks --- I think it would be slightly better to only regenerate the tabs if the previous value was a hidden tab.

Reply:
done in https://github.com/google/neuroglancer/pull/395/commits/cede0520e9c70ad77228b859ce8c5a8c2a660088

Reply:
forgot to update the prev variable, fixed in https://github.com/google/neuroglancer/commit/bc2a8d7a4980e5ce50832ba55f660619e6445bc8